packages/block-editor/src/components/url-input/index.js

```diff
@@ -58,6 +59,24 @@ class URLInput extends Component {
 				this.scrollingIntoView = false;
 			}, 100 );
 		}
+
+		// If there is no search text and no current suggestions
+		// then display the initial suggesitons if provided
+		// (being careful to avoid infinite re-render loop).
+		if ( __experimentalShowInitialSuggestions && ! ( value && value.length ) && ! ( suggestions && suggestions.length ) ) {
```

Comment:
This doesn't seem to work for me as expected. When adding a new link I see the loading spinner and endless requests being sent off

Reply:
It "works" but under the hood it's a (nearly) infinite loop. I'm fixing it.

Reply:
It "works" but under the hood it's a (nearly) infinite loop. I'm fixing it.

Reply:
This has now been fixed. I've added a flag within the `updateSuggestions()` call which determines whether an existing call to `updateSuggestions()` is already underway. We then use this (and other conditions) to conditionalise the fetching of initial suggestions on mount/update.

The issue was that `updateSuggestions` call in `componentDidUpdate` was firing repeatedly because the calls to `setState` within `updateSuggestions` trigger a re-render and thus triggered an infinite loop around `componentDidUpdate`!